src/VizTab.tsx

```diff
@@ -146,35 +153,72 @@ export default function VizTab(props: { namespace: string }) {
   // build base_link transform using Localization2DMsg
   useRobofleetMsgListener(
     matchTopic(props.namespace, 'localization'),
-    useCallback((buf, match) => {
-      const loc = fb.amrl_msgs.Localization2DMsg.getRootAsLocalization2DMsg(
-        buf
-      );
-      setX(loc.pose()?.x() ?? 0);
-      setY(loc.pose()?.y() ?? 0);
-      setTheta(loc.pose()?.theta() ?? 0);
-      setMapName(loc.map() ?? 'EmptyMap');
-      setLocAlertOpen(false);
-    }, [])
+    useCallback(
+      (buf, match) => {
+        const loc = fb.amrl_msgs.Localization2DMsg.getRootAsLocalization2DMsg(
+          buf
+        );
+        setX(loc.pose()?.x() ?? 0);
+        setY(loc.pose()?.y() ?? 0);
+        setTheta(loc.pose()?.theta() ?? 0);
+        setMapName(loc.map() ?? mapName);
```

Comment:
aren't most robots sending localization continuously? It seems like this would overwrite the value chosen by the user before they had a chance to set pose. Maybe add a separate state for user-selected map, and then send that when you set pose instead? 
In my mind, selecting a map would ideally just instantly send a "set map" command to the robot, but right now there's no way to do that without sending a pose too I think.

could be completely off base here, lmk

Reply:
Yeah, this is a fair point. I think the 2-variable approach is valid. The other one is to send an instant full pose message to the robot with whatever the current pose estimate is, but that kinda sucks